src/core/frame/names.cc

```diff
@@ -353,11 +353,22 @@ void Frame::_init_names(XTypeMaker& xt) {
 
 static const char * doc_options_frame_names_auto_index =
 R"(
-When Frame needs to auto-name columns, they will be assigned
-names `C0`, `C1`, `C2`, etc. by default. This option allows you to
-control the starting index in this sequence. For example, setting
-`dt.options.frame.names_auto_index=1` will cause the columns to be
-named `C1`, `C2`, `C3`, etc.
+
+This option controls the starting index that is used for auto-naming
+the columns. By default, the names that datatable assigns to frame's columns are
+`C0`, `C1`, `C2`, etc. Setting `names_auto_index`, for instance, to
+`1` will cause the columns to be named as `C1`, `C2`, `C3`, etc.
+
+Parameters
+----------
+return: int
+    Current `names_auto_index` value. Initially, this option is set to `0`.
+
+new_names_auto_index: int
+    New `names_auto_index` value.
+
+except: TypeError
+    The exception is raised when the type of `new_names_auto_index` is not `int`.
```

Comment:
This exception is raised for every single argument of every single function if you pass the incorrect type. So, I'm not sure this is something even worth documenting.

Reply:
That's true, not sure if it hurts though... especially that we already have it in many places.

Reply:
Well, it creates a source of inconsistency: if we document this for one function but not for other, then it looks like the other function does not check the types of its arguments?

Reply:
Well, if it looks like that then it's not because it is documented here, as here we do not report anything wrong, but because it is not documented for the other functions or for datatable in general. I feel like we need to create a dedicated page for all the possible types of exceptions/warnings datatable may produce and then remove all the individual `TypeError`s of this type everywhere in the docs.